crates/sui-indexer-alt-graphql/src/api/types/object.rs

```diff
@@ -234,33 +244,83 @@ impl Object {
         address: SuiAddress,
         version: UInt53,
     ) -> Result<Option<Self>, RpcError<Error>> {
-        let contents = ObjectContents::fetch(ctx, address, version).await?;
-        let Some(object) = &contents.0 else {
+        let Some(object) = contents(ctx, address, version).await? else {
             return Ok(None);
         };
 
         Ok(Some(Object {
             address: object.id().into(),
             version: object.version(),
             digest: object.digest(),
-            contents,
+            contents: Some(object),
```

Comment:
Checking my understanding: I see that we fill this field only when a user directly provides the version and loading is needed in this case to check for existence at the provided version. We lazily load the contents in all other cases because the server computes the version itself in all other cases via `version_bounded` and `checkpoint_bounded` where the version is determined by our PG tables, there's no need to consult kv for existence.